### PR TITLE
[SPARK-24646][CORE]  Minor change to spark.yarn.dist.forceDownloadSchemes to support wildcard '*'

### PR DESCRIPTION
**Pull request by jerryshao**
## What changes were proposed in this pull request?

In the case of getting tokens via customized `ServiceCredentialProvider`, it is required that `ServiceCredentialProvider` be available in local spark-submit process classpath. In this case, all the configured remote sources should be forced to download to local. 

For the ease of using this configuration, here propose to add wildcard '*' support to `spark.yarn.dist.forceDownloadSchemes`, also clarify the usage of this configuration.

## How was this patch tested?

New UT added.

### CODE REVIEW

**Comment by jerryshao** [comment]
Jenkins, retest this please.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 1]
```diff
@@ -483,10 +483,11 @@ package object config {
 
   private[spark] val FORCE_DOWNLOAD_SCHEMES =
     ConfigBuilder("spark.yarn.dist.forceDownloadSchemes")
-      .doc("Comma-separated list of schemes for which files will be downloaded to the " +
+      .doc("Comma-separated list of schemes for which resources will be downloaded to the " +
         "local disk prior to being added to YARN's distributed cache. For use in cases " +
         "where the YARN service does not support schemes that are supported by Spark, like http, " +
-        "https and ftp.")
+        "https and ftp, or jars required to add to local YARN client's classpath. Wildcard '*' is" +
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: "required to be in the local..."

Also, keep the space at the end of this line, like done in others,

File: docs/running-on-yarn.md [hunk 2]
```diff
@@ -218,9 +218,10 @@ To use a custom metrics.properties for the application master and executors, upd
   <td><code>spark.yarn.dist.forceDownloadSchemes</code></td>
   <td><code>(none)</code></td>
   <td>
-    Comma-separated list of schemes for which files will be downloaded to the local disk prior to
+    Comma-separated list of schemes for which resources will be downloaded to the local disk prior to
     being added to YARN's distributed cache. For use in cases where the YARN service does not
-    support schemes that are supported by Spark, like http, https and ftp.
+    support schemes that are supported by Spark, like http, https and ftp, or jars required to add to
```
Comment (vanzin) [inline docs/running-on-yarn.md]: Same slight reword here.

**Review by vanzin — COMMENTED** [review]
Looks good, just a couple of minor things.

**Comment by jerryshao** [comment]
Jenkins, retest this please.

**Comment by jerryshao** [comment]
Jenkins, retest this please.

**Comment by jerryshao** [comment]
Any more comment @vanzin ?

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Comment by jerryshao** [comment]
Thanks @jiangxb1987 , merging to master branch.

**Pull request closed without merging**